math.go

```diff
@@ -14,12 +17,12 @@ func dct(width int, height int, values []uint8) (result []float32) {
 				for j := 0; j < width; j++ {
 
 					sum += float64(values[i*width+j]) *
-						math.Cos(((math.Pi*float64(u))/(2*float64(height)))*(2*float64(i)+1)) *
-						math.Cos(((math.Pi*float64(v))/(2*float64(width)))*(2*float64(j)+1))
+						math.Cos(((math.Pi*float64(u))/doubleHeight)*(2*float64(i)+1)) *
+						math.Cos(((math.Pi*float64(v))/doubleWidth)*(2*float64(j)+1))
 				}
 			}
 
-			result[u*width+v] = float32(sum)
+			result[u*width+v] = int16(sum)
```

Comment:
Why are the result no longer `float`? I am looking at the example of DCT II value that is shown below the DCT table picture in this [IDCT example][dct-value]. The values in the matrix look like floats... and I would think that conterting them to `int16` would remove most of the information. :thinking:  What am I missing?

  [dct-value]: https://en.wikipedia.org/wiki/Discrete_cosine_transform#Example_of_IDCT